In [1]:
import pandas as pd
import numpy as np
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", None)
HC = pd.read_excel("ASHTURR5.xlsx")
SC = pd.read_excel("ASGTURR5.xlsx")

In [2]:
HC.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6032 entries, 0 to 6031
Columns: 118 entries, IDCNTRY to SCOPE
dtypes: float64(108), int64(10)
memory usage: 5.4 MB


HC is the home context data. This is a questionnaire filled in by the parents of each student. It contains crucial information on the immigration status of students and the languages they speak. 

In [3]:
SC.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6032 entries, 0 to 6031
Columns: 155 entries, IDCNTRY to SCOPE
dtypes: float64(129), int64(26)
memory usage: 7.1 MB


SC is the Student Context data which is a questionnaire filled in by the students themselves as well as some totals on their achievement in the test. As we can see there are 6032 children who took part in this assessment in Turkey. 

Let's look at the ages these children came to the country. ASBH02B 1: Younger than 3 years old; 2: 3 to 5 years old; 3: 6 to 7 years old; 4: 8 years old or older

In [4]:
number_immigrant_children = HC[HC["ASBH02B"]< 6.0].shape[0]
print(number_immigrant_children)

227


This is about 30 more than Germany but the Turkey sample size is bigger - 6032 kids, compared to 4611 in Germany. 

Now I will merge the two datasets together for fuller comparison potential.

In [6]:
df = pd.merge(HC, SC, on=None)

In [7]:
df['ASBH02B'].value_counts()

6.0    5366
1.0      89
2.0      70
3.0      53
9.0      31
4.0      15
Name: ASBH02B, dtype: int64

How many children spoke Turkish before they started school? ASBH03A (1 is yes, 2 is no)

In [8]:
df['ASBH03A'].value_counts()

1.0    5080
2.0     461
9.0      83
Name: ASBH03A, dtype: int64

In [9]:
def populate_column(row):
    if row['ASBH02B'] == 1:
        return 0
    elif row['ASBH02B'] == 2:
        return 3
    elif row['ASBH02B'] == 3:
        return 6
    elif row['ASBH02B'] == 4:
        return 8
    else:
        return None  # Or any default value

# Apply the function to each row
df['MINAGEARRIVAL'] = df.apply(populate_column, axis=1)

In [10]:
df['MINAGEARRIVAL'].value_counts()

0.0    89
3.0    70
6.0    53
8.0    15
Name: MINAGEARRIVAL, dtype: int64

In [11]:
df['ASDAGE'].round().value_counts()

10.0    5092
9.0      535
11.0     369
12.0      27
14.0       4
8.0        2
13.0       2
99.0       1
Name: ASDAGE, dtype: int64

In [12]:
df['rounded_ages'] = df['ASDAGE'].round()

We want to find the arabic speaking children now

In [13]:
df.columns

Index(['IDCNTRY', 'IDPOP', 'IDGRADER', 'IDGRADE', 'WAVE', 'IDSCHOOL',
       'IDCLASS', 'IDSTUD', 'ITLANG_HQ', 'LCID_HQ',
       ...
       'ASBGDRL', 'ASDGDRL', 'ASBGSLR', 'ASDGSLR', 'ASBGSCR', 'ASDGSCR',
       'ASDG05S', 'ASDRLOWP', 'MINAGEARRIVAL', 'rounded_ages'],
      dtype='object', length=263)

In [14]:
df.columns.to_list()

['IDCNTRY',
 'IDPOP',
 'IDGRADER',
 'IDGRADE',
 'WAVE',
 'IDSCHOOL',
 'IDCLASS',
 'IDSTUD',
 'ITLANG_HQ',
 'LCID_HQ',
 'ASBH01A',
 'ASBH01B',
 'ASBH01C',
 'ASBH01D',
 'ASBH01E',
 'ASBH01F',
 'ASBH01G',
 'ASBH01H',
 'ASBH01I',
 'ASBH01J',
 'ASBH01K',
 'ASBH01L',
 'ASBH01M',
 'ASBH01N',
 'ASBH01O',
 'ASBH01P',
 'ASBH01Q',
 'ASBH01R',
 'ASBH02A',
 'ASBH02B',
 'ASBH03A',
 'ASBH03B',
 'ASBH03C',
 'ASBH03D',
 'ASBH03E',
 'ASBH03F',
 'ASBH04',
 'ASBH05AA',
 'ASBH05AB',
 'ASBH05B',
 'ASBH06',
 'ASBH07A',
 'ASBH07B',
 'ASBH07C',
 'ASBH07D',
 'ASBH07E',
 'ASBH07F',
 'ASBH07G',
 'ASBH08A',
 'ASBH08B',
 'ASBH08C',
 'ASBH08D',
 'ASBH08E',
 'ASBH08F',
 'ASBH09',
 'ASBH10',
 'ASBH11A',
 'ASBH11B',
 'ASBH11C',
 'ASBH11D',
 'ASBH11E',
 'ASBH11F',
 'ASBH11G',
 'ASBH11H',
 'ASBH12',
 'ASBH13',
 'ASBH14A',
 'ASBH14B',
 'ASBH14C',
 'ASBH15A',
 'ASBH15B',
 'ASBH16',
 'ASBH17A',
 'ASBH17B',
 'ASBH18AA',
 'ASBH18AB',
 'ASBH18BA',
 'ASBH18BB',
 'ASBH18CA',
 'ASBH18CB',
 'ASBH18DA',
 'ASBH18DB',
 'ASBH18EA',
 '